rs/ic_os/config/src/main.rs

```diff
@@ -67,6 +72,13 @@ pub fn main() -> Result<()> {
             let deployment_json = read_deployment_file(&deployment_json_path)?;
             network_settings.mgmt_mac = deployment_json.deployment.mgmt_mac;
 
+            // get firewall.json rules
+            let firewall = get_firewall_rules_json_or_default(
+                firewall_json_path.as_ref().map(Path::new),
+                Path::new(config::DEFAULT_SETUPOS_FIREWALL_JSON_PATH),
+            )?;
+            network_settings.firewall = firewall;
+
```

Comment:
You may want to add similar logic to hostos_tool/main.rs to update the network_settings, though I'm not sure it will matter for correctness sake. You'll see what I mean when you look at the file. (Eventually, once we're fully on the new config pipeline, hostos won't have to generate network_settings itself as that will be handled in the setupOS config)

Reply:
We don't need to do that because the corresponding command (for both HostOS and GuestOS) is `render_firewall_rules` (last command in that match clause).

Reply:
It *does* bother me that the only instantiator of NetworkSettings is that read_config_ini function, which by definition produces an illegal NetworkSettings because it's always incomplete after reading config.ini.

But that's gonna go away anyway.

Reply:
Yes, I want to look into refactoring the call to  read_config_ini 👍